lua/octo/window.lua

```diff
@@ -25,78 +24,84 @@ function M.create_centered_float(opts)
       max_line = math.max(#line, max_line)
     end
   end
-
   -- calculate window height/width
   local width, height
   if max_line > 0 then
     -- pre-defined content
-    width = math.min(vim_width * 0.9, max_line)
+    width = math.min(vim_width * 0.9, max_line + 4)
     if opts.header then
-      height = math.min(vim_height, 2 + #opts.content)
+      height = math.min(vim_height, 4 + 2 + #opts.content)
     else
-      height = math.min(vim_height, #opts.content)
+      height = math.min(vim_height, 4 + #opts.content)
     end
   else
     width = math.floor(vim_width * opts.x_percent)
     height = math.floor(vim_height * opts.y_percent)
   end
-
   -- calculate offsets
-  local x_offset = (vim_width - width) / 2
-  local y_offset = (vim_height - height) / 2
-  local header_winid
-  local header_win_opts = {border = {}}
+  local x_offset = math.floor((vim_width - width) / 2)
+  local y_offset = math.floor((vim_height - height) / 2)
+
+  -- outer win (header + border)
```

Comment:
Comment are meant to express important information about code that the code itself can't express itself. Else comments are a source of many issues for code maintenance (c.f. Robert C. Martin - Clean Code). What does this comment state what the code itself doesn't tell the reader already. If it is about framing code blocks which belong together, than this is strong hint it should be exported into its own function because it is on another abstraction level as the parent function.

Reply:
Yep, they are used to frame code blocks. I will see if it can be split into different functions. Thanks